eslint/i18n/custom-rules/i18n-no-paramless-arrow.js

```diff
@@ -51,10 +49,6 @@ module.exports = {
           const key = tp.key.value;
 
           if (tp.value.type === 'ArrowFunctionExpression' && tp.value.params.length === 0) {
-            if (tp.value.body.type === 'TemplateLiteral') {
-              report(context, key, tp.value);
-            }
-
```

Comment:
This is a tricky one, feel free to tell me if it's unclear or if I'm wrong / forgot about something!

Before the fix, the rule was always flagging 2 errors instead of 1 for the same issue.
It was not a big deal and did not cause any issue, it wasn't visible in IDEs (I guess the LSP merged similar diagnostics for a single line but this is pure speculation).

It became visible when running automated tests (screenshot of unit tests running with the same code as master):
![screenshot of unit tests running with the same code as master, showing the tests fail because they received two errors instead of one](https://github.com/user-attachments/assets/043a66b5-32d5-4067-8050-807985636a8d)

As you can see, you get 2 errors about the same line & same column.

Apparently the `report` function does not stop the code from running so you have to be really careful with your conditions to avoid reporting the same error several times.

The following code:
```js
if (tp.value.body.type === 'TemplateLiteral') {
  report(context, key, tp.value);
}
```

handles this case:
![image](https://github.com/user-attachments/assets/c2eb929b-fbe6-4855-aace-86de4d13cf6c)

But since the `report` does not stop return / stop the function execution, we also run this code 

```js
if (!isSanitizeTagFunction(tp.value.body)) {
  report(context, key, tp.value);
}
```

This reports a second time because a `TemplateLiteral` Node can never be a `sanitize\`...\`` (`sanitize` is a `TaggedTemplateExpression` Node).

So I guess I could have kept the first test (templateLiteral) and removed the second one (isSanitize) but I felt the TemplateLiteral test was not enough, for instance, keeping only the `TemplateLiteral` test would mean that this:
```js
() => "Simple text"
```
would be valid because it's not a `TemplateLiteral`, it's a `Literal`:
![image](https://github.com/user-attachments/assets/c324e7a8-8ed9-4270-bed2-cfe478a75bdb)

Still we can agree that the paramless arrow is useless here too so I figured it was better to remove this test and keep only the important one "Does the arrow function contain a sanitize?" (yes => valid, no => invalid)

The `TemplateLiteral` test seems very redundant. I suspected the goal was to "early return" but for it to work we would have to properly "return". I figured it wasn't worth to try to early return because the `isSanitizeTagFunction` is very small / light :man_shrugging:.

Reply:
Thank you very much for the detailed explanations! :heart: 

It's very clear, removing this case totally makes sense to me now.